src/Streamly/Internal/Data/Stream/IsStream/Expand.hs

```diff
@@ -0,0 +1,816 @@
+import Streamly.Internal.Data.Stream.Async (async, wAsync)
+import Streamly.Internal.Data.Stream.IsStream.Common
+    ( concatM
+    , concatMapM
+    , concatMap
+    , smapM
+    , yield
+    , yieldM)
+import Streamly.Internal.Data.Stream.Parallel (parallel)
+import Streamly.Internal.Data.Stream.Prelude
+       ( concatFoldableWith, concatMapFoldableWith
+       , concatForFoldableWith, fromStreamS, toStreamS)
+import Streamly.Internal.Data.Stream.Serial (serial, wSerial)
+import Streamly.Internal.Data.Stream.StreamD (fromStreamD, toStreamD)
+import Streamly.Internal.Data.Stream.StreamK (IsStream)
+import Streamly.Internal.Data.SVar (MonadAsync)
+import Streamly.Internal.Data.Unfold.Type (Unfold)
+
+import qualified Streamly.Internal.Data.Stream.Parallel as Par
+import qualified Streamly.Internal.Data.Stream.Serial as Serial
+import qualified Streamly.Internal.Data.Stream.StreamD as D
+import qualified Streamly.Internal.Data.Stream.StreamK as K
+#ifdef USE_STREAMK_ONLY
+import qualified Streamly.Internal.Data.Stream.StreamK as S
+#else
+import qualified Streamly.Internal.Data.Stream.StreamD as S
+#endif
+import qualified Streamly.Internal.Data.Stream.Zip as Z
+
+import Prelude hiding (concat, concatMap)
+
+-- $setup
+-- >>> :m
+-- >>> import Prelude hiding (zipWith, concatMap, concat)
+-- >>> import qualified Streamly.Prelude as Stream
+-- >>> import Streamly.Internal.Data.Stream.IsStream as Stream
+-- >>> import qualified Streamly.Data.Fold as Fold
+-- >>> import qualified Streamly.Internal.Data.Fold as Fold
+-- >>> import qualified Streamly.Internal.Data.Unfold as Unfold
+-- >>> import qualified Streamly.Internal.Data.Parser as Parser
+-- >>> import qualified Streamly.Data.Array.Foreign as Array
+
+-------------------------------------------------------------------------------
+-- Appending
+-------------------------------------------------------------------------------
+
+-- XXX Reconcile the names "serial" and "append".
+--
+-- | Append the outputs of two streams, yielding all the elements from the
+-- first stream and then yielding all the elements from the second stream.
+--
+-- IMPORTANT NOTE: This could be 100x faster than @serial/<>@ for appending a
+-- few (say 100) streams because it can fuse via stream fusion. However, it
+-- does not scale for a large number of streams (say 1000s) and becomes
+-- qudartically slow. Therefore use this for custom appending of a few streams
+-- but use 'concatMap' or 'concatMapWith serial' for appending @n@ streams or
+-- infinite containers of streams.
+--
+-- /Pre-release/
+{-# INLINE append #-}
+append ::(IsStream t, Monad m) => t m b -> t m b -> t m b
+append m1 m2 = fromStreamD $ D.append (toStreamD m1) (toStreamD m2)
+
+-------------------------------------------------------------------------------
+-- Interleaving
+-------------------------------------------------------------------------------
+
+-- XXX Same as 'wSerial'. We should perhaps rename wSerial to interleave.
```

Comment:
I would appreciate the other way around.
`append` ~> `serialD`
`interleave` ~> `wSerialD`
or something similar.